Source/SwiftLintFramework/Rules/Style/ExplicitReturnRule.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import SourceKittenFramework
+import SwiftSyntax
+
+public struct ExplicitReturnRule: ConfigurationProviderRule, SubstitutionCorrectableRule, OptInRule {
+    public var configuration = ExplicitReturnConfiguration()
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "explicit_return",
+        name: "Explicit Return",
+        description: "Prefer explicit returns in closures, functions and getters.",
+        kind: .style,
+        nonTriggeringExamples: ExplicitReturnRuleExamples.nonTriggeringExamples,
+        triggeringExamples: ExplicitReturnRuleExamples.triggeringExamples,
+        corrections: ExplicitReturnRuleExamples.corrections
+    )
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        return violationOffsets(file: file).map {
+            StyleViolation(
+                ruleDescription: Self.description,
+                severity: configuration.severityConfiguration.severity,
+                location: Location(file: file, byteOffset: $0)
+            )
+        }
+    }
+
+    public func violationRanges(in file: SwiftLintFile) -> [NSRange] {
+        return violationOffsets(file: file).compactMap {
+            file.stringView.byteRangeToNSRange(ByteRange(location: $0, length: 0))
+        }
+    }
+
+    public func substitution(for violationRange: NSRange, in file: SwiftLintFile) -> (NSRange, String)? {
+        return (violationRange, "return ")
+    }
+
+    private func violationOffsets(file: SwiftLintFile) -> [ByteCount] {
+        guard let tree = file.syntaxTree else { return [] }
+
+        let visitors: [ViolationReporting] = [
+            ClosureExprSyntaxVisitor(),
+            FunctionDeclSyntaxVisitor(),
+            VariableDeclSyntaxVisitor()
+        ].filter {
+            configuration.isKindIncluded($0.kind)
+        }
+
+        for visitor in visitors {
+            visitor.walk(tree)
+        }
+
+        return visitors
+            .reduce(into: [], { $0.append(contentsOf: $1.positions) })
+            .map { ByteCount($0.utf8Offset) }
```

Comment:
if you don't combine the visitor into one, you should add a `sorted()` here since we prefer that the violations are sorted by offset (I think this is encoded in our unit tests, you probably just don't have an example with 2+ violations)

Reply:
Due to merging visitors into a single one, I consider this one done.